File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/di/DomainModule.java
package space.lopatkin.spb.helpboardgamecard.di;

import dagger.Module;
import dagger.Provides;
import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.DeleteHelpcardUnlockedByHelpcardIdUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.DeleteHelpcardUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.DeleteHelpcardsByUnlockStateUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.GetAllHelpcardsUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.GetHelpcardByHelpcardIdUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.GetKeyboardTypeUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.SaveHelpcardNewByHelpcardIdUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.SaveKeyboardTypeByUserChoiceUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.UpdateHelpcardByHelpcardIdUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.UpdateHelpcardFavoriteByHelpcardIdUseCase;
import space.lopatkin.spb.helpboardgamecard.domain.usecase.UpdateHelpcardLockingByHelpcardIdUseCase;

import javax.inject.Singleton;

@Module
public class DomainModule {

    @Singleton
    @Provides
    public GetHelpcardByHelpcardIdUseCase provideGetHelpcardByHelpcardIdUseCase(AppRepository repository) {
        return new GetHelpcardByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public DeleteHelpcardUseCase provideDeleteHelpcardUseCase(AppRepository repository) {
        return new DeleteHelpcardUseCase(repository);
    }

    @Singleton
    @Provides
    public DeleteHelpcardUnlockedByHelpcardIdUseCase provideDeleteHelpcardUnlockedByHelpcardIdUseCase(AppRepository repository) {
        return new DeleteHelpcardUnlockedByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public GetAllHelpcardsUseCase provideGetAllHelpcardsUseCase(AppRepository repository) {
        return new GetAllHelpcardsUseCase(repository);
    }

    @Singleton
    @Provides
    public UpdateHelpcardByHelpcardIdUseCase provideUpdateHelpcardByHelpcardIdUseCase(AppRepository repository) {
        return new UpdateHelpcardByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public UpdateHelpcardFavoriteByHelpcardIdUseCase provideUpdateHelpcardFavoriteByHelpcardIdUseCase(AppRepository repository) {
        return new UpdateHelpcardFavoriteByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public UpdateHelpcardLockingByHelpcardIdUseCase provideUpdateHelpcardLockingByHelpcardIdUseCase(AppRepository repository) {
        return new UpdateHelpcardLockingByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public DeleteHelpcardsByUnlockStateUseCase provideDeleteHelpcardsByUnlockStateUseCase(AppRepository repository) {
        return new DeleteHelpcardsByUnlockStateUseCase(repository);
    }

    @Singleton
    @Provides
    public SaveHelpcardNewByHelpcardIdUseCase provideSaveHelpcardNewByHelpcardIdUseCase(AppRepository repository) {
        return new SaveHelpcardNewByHelpcardIdUseCase(repository);
    }

    @Singleton
    @Provides
    public SaveKeyboardTypeByUserChoiceUseCase provideSaveKeyboardTypeByUserChoiceUseCase(AppRepository repository) {
        return new SaveKeyboardTypeByUserChoiceUseCase(repository);
    }

    @Singleton
    @Provides
    public GetKeyboardTypeUseCase provideGetKeyboardTypeUseCase(AppRepository repository) {
        return new GetKeyboardTypeUseCase(repository);
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/di/DomainModule.kt
package space.lopatkin.spb.helpboardgamecard.di

import dagger.Module
import dagger.Provides
import space.lopatkin.spb.helpboardgamecard.domain.repository.BoardgameRepository
import space.lopatkin.spb.helpboardgamecard.domain.repository.SettingsRepository
import space.lopatkin.spb.helpboardgamecard.domain.usecase.*
import javax.inject.Singleton

@Module
class DomainModule {

    @Singleton
    @Provides
    fun provideGetHelpcardByBoardgameIdUseCase(repository: BoardgameRepository): GetHelpcardByBoardgameIdUseCase {
        return GetHelpcardByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideDeleteBoardgameUnlockedByBoardgameIdUseCase(repository: BoardgameRepository): DeleteBoardgameUnlockedByBoardgameIdUseCase {
        return DeleteBoardgameUnlockedByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideGetAllBoardgamesInfoUseCase(repository: BoardgameRepository): GetAllBoardgamesInfoUseCase {
        return GetAllBoardgamesInfoUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideUpdateBoardgameByBoardgameIdUseCase(repository: BoardgameRepository): UpdateBoardgameByBoardgameIdUseCase {
        return UpdateBoardgameByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideUpdateBoardgameFavoriteByBoardgameIdUseCase(repository: BoardgameRepository): UpdateBoardgameFavoriteByBoardgameIdUseCase {
        return UpdateBoardgameFavoriteByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideUpdateBoardgameLockingByBoardgameIdUseCase(repository: BoardgameRepository): UpdateBoardgameLockingByBoardgameIdUseCase {
        return UpdateBoardgameLockingByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideDeleteBoardgamesByUnlockStateUseCase(repository: BoardgameRepository): DeleteBoardgamesByUnlockStateUseCase {
        return DeleteBoardgamesByUnlockStateUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideSaveBoardgameNewByBoardgameIdUseCase(repository: BoardgameRepository): SaveBoardgameNewByBoardgameIdUseCase {
        return SaveBoardgameNewByBoardgameIdUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideSaveKeyboardTypeByUserChoiceUseCase(repository: SettingsRepository): SaveKeyboardTypeByUserChoiceUseCase {
        return SaveKeyboardTypeByUserChoiceUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideGetKeyboardTypeUseCase(repository: SettingsRepository): GetKeyboardTypeUseCase {
        return GetKeyboardTypeUseCase(repository = repository)
    }

    @Singleton
    @Provides
    fun provideGetBoardgameRawByBoardgameIdUseCase(repository: BoardgameRepository): GetBoardgameRawByBoardgameIdUseCase {
        return GetBoardgameRawByBoardgameIdUseCase(repository = repository)
    }

}
